<launch>
    <rosparam command="load" file="$(find object_tracking)/config/config.yaml" />

    <!-- Realsense ros-->
    <arg name="serial_no"           default=""/>
    <arg name="json_file_path"      default=""/>
    <arg name="camera"              default="camera"/>
    <arg name="tf_prefix"           default="$(arg camera)"/>

    <arg name="fisheye_width"       default="640"/>
    <arg name="fisheye_height"      default="480"/>
    <arg name="enable_fisheye"      default="true"/>

    <arg name="depth_width"         default="640"/>
    <arg name="depth_height"        default="480"/>
    <arg name="enable_depth"        default="true"/>

    <arg name="infra_width"         default="640"/>
    <arg name="infra_height"        default="480"/>
    <arg name="enable_infra1"       default="true"/>
    <arg name="enable_infra2"       default="true"/>

    <arg name="color_width"         default="640"/>
    <arg name="color_height"        default="480"/>
    <arg name="enable_color"        default="true"/>

    <arg name="fisheye_fps"         default="30"/>
    <arg name="depth_fps"           default="30"/>
    <arg name="infra_fps"           default="30"/>
    <arg name="color_fps"           default="30"/>
    <arg name="gyro_fps"            default="400"/>
    <arg name="accel_fps"           default="250"/>
    <arg name="enable_gyro"         default="true"/>
    <arg name="enable_accel"        default="true"/>

    <arg name="enable_pointcloud"   default="false"/>
    <arg name="enable_sync"         default="true"/>
    <arg name="align_depth"         default="true"/>

    <arg name="filters"             default=""/>

    <group ns="$(arg camera)">
        <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
        <arg name="tf_prefix"                value="$(arg tf_prefix)"/>
        <arg name="serial_no"                value="$(arg serial_no)"/>
        <arg name="json_file_path"           value="$(arg json_file_path)"/>

        <arg name="enable_pointcloud"        value="$(arg enable_pointcloud)"/>
        <arg name="enable_sync"              value="$(arg enable_sync)"/>
        <arg name="align_depth"              value="$(arg align_depth)"/>

        <arg name="fisheye_width"            value="$(arg fisheye_width)"/>
        <arg name="fisheye_height"           value="$(arg fisheye_height)"/>
        <arg name="enable_fisheye"           value="$(arg enable_fisheye)"/>

        <arg name="depth_width"              value="$(arg depth_width)"/>
        <arg name="depth_height"             value="$(arg depth_height)"/>
        <arg name="enable_depth"             value="$(arg enable_depth)"/>

        <arg name="color_width"              value="$(arg color_width)"/>
        <arg name="color_height"             value="$(arg color_height)"/>
        <arg name="enable_color"             value="$(arg enable_color)"/>

        <arg name="infra_width"              value="$(arg infra_width)"/>
        <arg name="infra_height"             value="$(arg infra_height)"/>
        <arg name="enable_infra1"            value="$(arg enable_infra1)"/>
        <arg name="enable_infra2"            value="$(arg enable_infra2)"/>

        <arg name="fisheye_fps"              value="$(arg fisheye_fps)"/>
        <arg name="depth_fps"                value="$(arg depth_fps)"/>
        <arg name="infra_fps"                value="$(arg infra_fps)"/>
        <arg name="color_fps"                value="$(arg color_fps)"/>
        <arg name="gyro_fps"                 value="$(arg gyro_fps)"/>
        <arg name="accel_fps"                value="$(arg accel_fps)"/>
        <arg name="enable_gyro"              value="$(arg enable_gyro)"/>
        <arg name="enable_accel"             value="$(arg enable_accel)"/>

        <arg name="filters"                  value="$(arg filters)"/>
        </include>
    </group>


    <!-- Aruco node-->
    <arg name="markerId"        default="6"/>
    <arg name="markerSize"      default="0.04"/>    <!-- in m -->
    <arg name="marker_frame"    default="aruco_marker_frame"/>
    <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->
	<arg name="ns_camera" default="camera"/>

    <node pkg="aruco_ros" type="single" name="aruco_single">
		<remap from="/camera_info" to="/$(arg ns_camera)/color/camera_info" />
		<remap from="/image" to="/$(arg ns_camera)/color/image_raw" />
        <param name="image_is_rectified" value="True"/>
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
		<param name="camera_frame"       value="$(arg ns_camera)_color_optical_frame"/>
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>

    <!--node pkg="object_tracking" type="matching_node.py" name="matcher"/ -->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find object_tracking)/launch/rviz_launch.rviz" />
</launch>